sound/soc/sof/topology.c

```diff
@@ -2092,6 +2171,30 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 
 	sof_dbg_comp_config(scomp, &process->config);
 
+	/* possible tuplet and data parsing for process components */
+	switch (type) {
+	case SOF_COMP_EQ_FIR:
+		/* fallthrough */
+	case SOF_COMP_EQ_IIR:
+		/* fallthrough */
+	case SOF_COMP_KEYWORD_DETECT:
+		/* fallthrough */
+	case SOF_COMP_KPB:
+		/* fallthrough */
+	case SOF_COMP_SELECTOR:
+		break;
+	case SOF_COMP_MUX:
+	case SOF_COMP_DEMUX:
+		offset = sof_process_load_mux(scomp, process, private);
+		break;
+	}
+
+	if (offset < 0) {
+		dev_err(scomp->dev, "error: parse process tokens failed %d\n",
+			le32_to_cpu(private->size));
+		goto err;
+	}
```

Comment:
better move this inside the `switch` - only there `offset` can be negative - right after the assignment.

Reply:
ok